packages/edit-post/src/components/sidebar/sidebar-header/index.js

```diff
@@ -23,6 +23,14 @@ const SidebarHeader = ( { children, className, closeLabel } ) => {
 	);
 	const { closeGeneralSidebar } = useDispatch( 'core/edit-post' );
 
+	// The `tabIndex` serves the purpose of normalizing browser behavior of
+	// button clicks and focus. Notably, without making the header focusable, a
+	// Button click would not trigger a focus event in macOS Firefox. Thus, when
+	// the sidebar is unmounted, the corresponding "focus return" behavior to
+	// shift focus back to the heading toolbar would not be run.
+	//
+	// See: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Clicking_and_focus
```

Comment:
I know we tried this before but forgot the conclusion, why don't we try to normalize it on the Button click handler (or related events).

Reply:
Asking cause it's not the first time we have the button click focus issues and certainly not the last time.

Reply:
I've been thinking about this as well.

I think one of the main reluctances I'd have is in _breaking expectations_ about how elements are expected to work, but I'm not sure how well this holds up:

- The abstraction of a button shouldn't necessarily be known to be tied to expectations around a particular HTML element.
- The "expected" behavior is clearly not very _expected_, or at least it's highly inconsistent between browsers and consequently prone to error.

On the technical front, I've been wondering as well just how this would work:

- If we dispatch a `new window.Event`, how well does would it interoperate with React's event system, and in particular with [portaled](https://reactjs.org/docs/portals.html) rendering which can't rely on DOM event bubbling.

I've also been thinking about a few other options:

- Is `<input type="button">` susceptible to this as well? And if not, could we implement `<Button>` this way? I expect by virtue of the differences between the two (largely ability to use non-text HTML content in `<button>`), it may be infeasible.
- What about something like `<span role="button">` ? As I understand it, there are a lot of accessibility challenges in using `role="button"` effectively, but if it were technically possible, we could absorb that complexity into the implementation of the core component.

Reply:
I think @diegohaz might have good thoughts on this problem as something he probably already faced with reakit

Reply:
On Reakit we opted to ensure consistency across browsers on a `Tabbable` component (which is used by `Button`): https://github.com/reakit/reakit/blob/c9f55d7d87f12b58ee409e8a392a9c36c2aadf52/packages/reakit/src/Tabbable/Tabbable.ts#L141-L148

It's important to note that, on Firefox/macOS, calling `preventDefault()` there also prevents it to get the `:active` state when clicked (https://github.com/reakit/reakit/issues/432). Something like `data-active` could be used instead. That's the only bad side-effect people noticed so far.

Reply:
@diegohaz Thanks for sharing! It hadn't occurred to me as potentially being as simple as calling `.focus()` on the button node. About the implementation and your point about `preventDefault`: Can you explain why `preventDefault` is necessary at all?

Reply:
That's because Safari/Firefox on macOS will blur the button if it has focus on mouse down. `preventDefault` prevents that from happening. But there may be another way to work around that.